junifer/configs/juseless/datagrabbers/ucla.py

```diff
@@ -0,0 +1,120 @@
+"""Provide a concrete implementation for UCLA dataset."""
+
+# Authors: Federico Raimondo <f.raimondo@fz-juelich.de>
+#          Leonard Sasse <l.sasse@fz-juelich.de>
+# License: AGPL
+
+from pathlib import Path
+from typing import List, Union
+
+from ....datagrabber import PatternDataGrabber
+from ....api.decorators import register_datagrabber
+from ....utils import raise_error
+
+
+@register_datagrabber
+class JuselessUCLA(PatternDataGrabber):
+    """Concrete implementation for pattern-based data fetching of UCLA data.
+
+    Parameters
+    ----------
+    datadir : str or Path, optional
+        The directory where the dataset is stored
+        (default '/data/project/psychosis_thalamus/data/fmriprep').
+    **kwargs
+        Keyword arguments passed to superclass.
+
+    """
+
+    def __init__(
+        self,
+        datadir: Union[
+            str, Path, None
```

Comment:
Does it make sense for a pattern-based datagrabber to have `None`? I think you always need to have `str` or `Path`.

Reply:
None is only for DataladDatagrabbers

Reply:
ok makes sense!